### PR TITLE
README updates: more examples, explain auth methods

### PR DESCRIPTION
**Pull request by natemccurdy**
#### Pull Request (PR) description
This updates the readme to improve it in the following ways:
* Adds examples of using the options hash vs. positional arguments
* Clarifies each available config option in both the options hash and positional args.
* Adds Details on supported authentication methods.
* Fixes a few typos and some inaccurate information.

This also restructures the README to have requirements first, usage and config examples next, then ends with details on the supported authentication methods.
I think this flow is better because people usually want to see setup info and example usage first. Prior to this, that was buried beneath details about the authentication methods.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,106 +15,38 @@ agents with Hashicorp Vault.
...
+Puppet module provides the `vault_lookup::lookup()` function.
+
+When used with Puppet 6's [`Deferred`
+type](https://puppet.com/docs/puppet/7/deferring_functions.html), the function
+allows agents to retrieve secrets from Vault when a catalog is applied rather
+than compiled. In this way, the secret data is not embedded in the catalog and
+the Puppetserver does not need permissions to read all your Vault secrets.
+
+See [this blog
 post](https://puppet.com/blog/secret-agents-man-secrets-store-integrations-puppet-6)
```
Comment (smortex) [inline README.md:35]: Broken link :disappointed: 

Available on the internet archive:
https://web.archive.org/web/20220520045735/https://www.puppet.com/blog/secret-agents-man-secrets-store-integrations-puppet-6/

Comment (natemccurdy) [reply README.md:35]: Aww, bummer.
Are we okay with linking to the web archive?

Comment (smortex) [reply README.md:35]: Not sure it's the best: to me it looks like promotion of Puppet 6 which was just announced, and now that Puppet 6 is EOL, it has low value maybe?  The current documentation should have all the content in a more up-to-date form I believe :shrug: .

Comment (natemccurdy) [reply README.md:35]: Yup, agreed that that blog post isn't really useful.
I'll either drop this line entirely, or find something more useful to link to.

Comment (bastelfreak) [reply README.md:35]: if you cannot find an alternative I think the link is fine. also CC @binford2k about the broken link

**Review by bastelfreak — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (natemccurdy) [reply README.md]: Decided to just drop the link.

**Pull request merged**